integration/pkg/aws/cleanup.go

```diff
@@ -13,6 +15,47 @@ import (
 	nfwTypes "github.com/aws/aws-sdk-go-v2/service/networkfirewall/types"
 )
 
+func (id *OnvIntegrationTestData) CleanupSecurityGroup(ctx context.Context) error {
+	resp, err := id.ec2Api.DescribeSecurityGroups(ctx, &ec2.DescribeSecurityGroupsInput{
+		Filters: securityGroupTagFilters(),
+	})
+	if err != nil {
+		return err
+	}
+
+	if len(resp.SecurityGroups) == 0 {
+		log.Println("No SecurityGroups found - skipping cleanup")
+		return nil
+	}
+
+	log.Println("WARN: Found a leaked osd-network-verifier security group. This is unexpected, as the verifier should fully clean up after itself. Cleaning up anyways...")
```

Comment:
@abyrne55 I added that message here, as this is the point at which we would have a non-zero number of security groups to clean up. This way, the message below still logs individual SGs that get deleted.